### PR TITLE
fix: use the correct field name when reading data from multi fields

### PR DESCRIPTION
**Pull request by salvatore-campagna**
When using a multi-field we need to extract data from the document
using the correct field name. That is the name of the top field.
Here we delegate extraction of the correct name to a method in the
SearchContext that is wrapped by the AggregationContext.

Issue: #82918

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/SignificantTextAggregatorFactory.java [hunk 1]
```diff
@@ -253,13 +259,15 @@ private CollectorSource createCollectorSource() {
             MappedFieldType fieldType,
             Analyzer analyzer,
             String[] sourceFieldNames,
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/SignificantTextAggregatorFactory.java:261]: I wonder if we could flatmap the sourceFieldNames in the code that calls the ctor.

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/470_significant_texts.yml [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+          - { "full_text": "elma muz armut" }
+          - { "index": { } }
+          - { "full_text": "elma armut kavun" }
+
+
+---
+"significant_texts all terms english":
+  - skip:
+      version: " - 8.1.99"
+      reason: bug fixed in 8.2.0
```
[the remark this answers is not in the document]
Comment (salvatore-campagna) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search.aggregation/470_significant_texts.yml:46]: Why `8.3`...master in `8.2` until March 29 right?

**Comment by salvatore-campagna** [comment]
@elasticmachine run elasticsearch-ci/part2

**Comment by salvatore-campagna** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by salvatore-campagna** [comment]
@elasticmachine run elasticsearch-ci/part-2

**Comment by salvatore-campagna** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/SignificantTextAggregatorFactory.java [hunk 3]
```diff
@@ -218,13 +219,17 @@ protected Aggregator createInternal(Aggregator parent, CardinalityUpperBound car
...
+            .toArray(String[]::new);
         if (context.profiling()) {
             return new ProfilingSignificantTextCollectorSource(
                 context.lookup().source(),
                 context.bigArrays(),
                 fieldType,
                 analyzer,
-                sourceFieldNames,
+                fieldNames,
+                context,
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/SignificantTextAggregatorFactory.java:232]: Do these need the context any more?

**Review by nik9000 — APPROVED** [review]
Great!

**Comment by salvatore-campagna** [comment]
@elasticmachine update branch

**Pull request merged**